SchedulePlatform.Api/Controllers/AppointmentController.cs

```diff
@@ -1,81 +1,170 @@
 ﻿using System;
 using Microsoft.AspNetCore.Mvc;
-using SchedulePlatform.Api.Models;
 using SchedulePlatform.Api.Mappings;
 using SchedulePlatform.Models.Entities;
 using SchedulePlatform.Service.Interfaces;
+using SchedulePlatform.Api.Models.Patch;
+using SchedulePlatform.Service.Models.Appointment;
+using SchedulePlatform.Service.Models;
+using AutoMapper;
 
 namespace SchedulePlatform.Api.Controllers
 {
-	[Route("api/[controller]")]
-
-	[ApiController]
-
-	public class AppointmentController : ControllerBase
-	{
-		private readonly IAppointmentService _service;
-
-		public AppointmentController(IAppointmentService service)
-		{
-			_service = service;
-		}
-
-		[HttpGet]
-
-		public Appointment[] GetAppointments()
-		{
-			return _service.GetAll();
-		}
-
-		[HttpPost]
-
-		public Appointment AddAppointment(Appointment appointment)
-		{
-			return _service.Add(appointment);
-		}
-
-		[HttpGet("{id}")]
-
-		public IActionResult GetById(Guid id)
-		{
-			var appointmentResult = _service.GetById(id);
-
-			if (appointmentResult==null)
-			{
-				return NotFound();
-			}
-			return Ok(appointmentResult);
-		}
-
-		[HttpPatch]
-
-		public IActionResult Update(Guid id, AppointmentPatchModel model)
-		{
-			var appointmentSearch = _service.GetById(id);
-
-			if (appointmentSearch==null)
-			{
-				return NotFound();
-			}
-
-			var appointmentUpdated = appointmentSearch.Map(model);
-
-			return Ok(_service.Update(appointmentUpdated));
-		}
-
-		[HttpDelete]
-
-		public IActionResult Delete(Guid id,Appointment appointment)
-		{
-			var appointmentSearch = _service.GetById(id);
-
-			if (appointmentSearch==null)
-			{
-				return NotFound();
-			}
-
-			return Ok(_service.Delete(id, appointment));
-		}
-	}
+    [Route("api/[controller]")]
+
+    [ApiController]
+
+    public class AppointmentController : ControllerBase
+    {
+        private readonly IAppointmentService _service;
+        private readonly IMapper _mapper;
+
+        public AppointmentController(IAppointmentService service,IMapper mapper)
+        {
+            _service = service;
+            _mapper = mapper;
+        }
+
+        [HttpGet("allApp/{nutritionistId}")]
```

Comment:
"allApp" should not be here. "{nutritionistId}" is just enough to have the route as "api/appointments/{nutritionistId}".

Reply:
Hello Nico, it doesn t allow me as i receive an error of: microsoft.AspNetCore.Routing.Matching.AmbiguousMatchException: The request matched multiple endpoints.
I guessed because I have 3 requests, one based on appointment's id, one on nutritionist id and the other on customer's id